### ISSUE TITLE
`citation-number` is treated as a string, not a number

### ISSUE THREAD

**Issue by slochower** [jgm/pandoc-citeproc#399]
It appears that `citation-number` is treated as a string and therefore sorted alphabetically (at least in some styles) like this `[10, 11, 8, 9]` instead of numeric order `[8, 9, 10, 11]` (or reverse, depending on style).

References: https://github.com/manubot/rootstock/issues/206

**Comment by jgm** [comment]
Can you give a full example that allows us to
reproduce this?  (Including bibliography, markdown,
and csl?)

**Comment by jgm** [comment]
For an example of a convenient format for a minimal
test case, see https://github.com/jgm/pandoc-citeproc/issues/393

**Comment by slochower** [comment]
Sure! A little longer than #393 because I need to go up to double digits to show the effect.

```
pandoc -F pandoc-citeproc -t plain << EOT

---
references:
- id: 'one'
  author:
    - family: One
- id: 'two'
  author:
    - family: Two
- id: 'three'
  author:
    - family: Three
- id: 'four'
  author:
    - family: Four
- id: 'five'
  author:
    - family: Five
- id: 'six'
  author:
    - family: Six
- id: 'seven'
  author:
    - family: Seven
- id: 'eight'
  author:
    - family: Eight
- id: 'nine'
  author:
    - family: Nine
- id: 'ten'
  author:
    - family: Ten
- id: 'eleven'
  author:
    - family: Eleven

csl: style.csl
---

Inline citation [@one; @two; @three; @four; @five; @six; @seven; @eight; @nine; @ten; @eleven].

EOT
```

With a minimal `style.csl` as follows:
```
<?xml version="1.0" encoding="utf-8"?>
<style xmlns="http://purl.org/net/xbiblio/csl" class="in-text" version="1.0" demote-non-dropping-particle="never" default-locale="en-US">
  <citation>
    <sort>
      <key variable="citation-number"/>
    </sort>
    <layout prefix="[" suffix="]" delimiter=",">
      <text variable="citation-number"/>
    </layout>
  </citation>
</style>
```

**Comment by jgm** [comment]
So, to confirm, this issue is about sort order in the citation, not in the bibliography, correct?

**Comment by slochower** [comment]
Correct. That style doesn't print a bibliography, but even when printing one using the same ordering as inline, the bibliography will correctly go from 1 to 11 while the inline citation does not.

**Comment by jgm** [comment]
Note to self: this happens in Text.CSL.Eval, evalSorting.

**Comment by jgm** [comment]
Argh, `Sorting` gets values like `Ascending "10"`; it's `Ascending String`.
We'll need to change these types.

**Comment by slochower** [comment]
Thanks! Once you cut a new release, I'll upgrade and check on my end. 👍

**Comment by jgm** [comment]
released 0.16.2 on Hackage

**Comment by slochower** [comment]
Working as expected for me. Thanks.
